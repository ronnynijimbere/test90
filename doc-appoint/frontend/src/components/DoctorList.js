import React from "react";
import { useNavigate } from "react-router-dom";
import { useSelector } from "react-redux";

const DoctorList = ({ doctor, handleAccountStatus }) => {
  const navigate = useNavigate(); // Hook to navigate programmatically
  const { user } = useSelector((state) => state.user); // Get user information from the Redux store

  // Function to handle card click
  const handleCardClick = () => {
    navigate(`/doctor/profile/${doctor._id}`); // Navigate to doctor's profile page
  };

  return (
    <div className="card m-2" style={{ cursor: "pointer" }} onClick={handleCardClick}>
      {/* Doctor's name in the card header */}
      <div className="card-header" style={{ textAlign: "center", fontWeight: "bold" }}>
        Dr. {doctor.firstName} {doctor.lastName}
      </div>
      <div className="card-body">
        {/* Display doctor's specialization */}
        <p><b>Specialization:</b> {doctor.specialization}</p>
        {/* Display doctor's experience */}
        <p><b>Experience:</b> {doctor.experience}</p>
        {/* Display doctor's fees per consultation */}
        <p><b>Fees Per Consultation:</b> {doctor.feesPerConsultation}</p>
        {/* Display doctor's available timings */}
        <p><b>Timings:</b> {doctor.starttime} - {doctor.endtime}</p>
        
        {/* Admin actions: Approve, Reject, or Block based on doctor's status */}
        {user.role === "admin" && (
          <div className="admin-actions">
            {doctor.status === "pending" ? (
              <>
                <button className="m-1 btn btn-success" onClick={() => handleAccountStatus(doctor, "approved")}>
                  Approve
                </button>
                <button className="m-1 btn btn-danger" onClick={() => handleAccountStatus(doctor, "rejected")}>
                  Reject
                </button>
              </>
            ) : (
              <button className="m-1 btn btn-danger" onClick={() => handleAccountStatus(doctor, "blocked")}>
                Block
              </button>
            )}
          </div>
        )}
      </div>
    </div>
  );
};

export default DoctorList;
