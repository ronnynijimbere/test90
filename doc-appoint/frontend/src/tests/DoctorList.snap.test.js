// DoctorList.snap.test.js

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import DoctorList from '../components/DoctorList';
import { BrowserRouter } from 'react-router-dom';

// Mock Redux store
const mockStore = configureMockStore();
const store = mockStore({
  user: {
    user: {
      role: "admin",
    },
  },
});

// Suppress console.error during tests
beforeAll(() => {
  jest.spyOn(console, 'error').mockImplementation(() => {});
});

// Restore console.error after tests
afterAll(() => {
  console.error.mockRestore();
});

// Mock doctor data
const doctor = {
  firstName: "John",
  lastName: "Doe",
  specialization: "Cardiology",
  experience: "10 years",
  feesPerConsultation: "$100",
  starttime: "09:00 AM",
  endtime: "05:00 PM",
  status: "pending",
};

// Mock handleAccountStatus function
const handleAccountStatus = jest.fn();

// Snapshot test for DoctorList component
it('renders DoctorList component correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <DoctorList doctor={doctor} handleAccountStatus={handleAccountStatus} />
      </BrowserRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot(); // Compare rendered component to the snapshot
});
















