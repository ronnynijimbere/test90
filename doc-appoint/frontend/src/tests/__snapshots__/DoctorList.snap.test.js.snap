// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DoctorList component correctly 1`] = `
<div
  className="card m-2"
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <div
    className="card-header"
    style={
      Object {
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Dr. 
    John
     
    Doe
  </div>
  <div
    className="card-body"
  >
    <p>
      <b>
        Specialization:
      </b>
       
      Cardiology
    </p>
    <p>
      <b>
        Experience:
      </b>
       
      10 years
    </p>
    <p>
      <b>
        Fees Per Consultation:
      </b>
       
      $100
    </p>
    <p>
      <b>
        Timings:
      </b>
       
      09:00 AM
       - 
      05:00 PM
    </p>
    <div
      className="admin-actions"
    >
      <button
        className="m-1 btn btn-success"
        onClick={[Function]}
      >
        Approve
      </button>
      <button
        className="m-1 btn btn-danger"
        onClick={[Function]}
      >
        Reject
      </button>
    </div>
  </div>
</div>
`;
